merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java

```diff
@@ -90,8 +111,223 @@ SimulationResults simulate(
         timeline.add(commit);
       }
 
-      return SimulationEngine.computeResults(engine, startTime, elapsedTime, activityTopic, timeline, missionModel);
+      return computeResults(engine, startTime, elapsedTime, activityTopic, timeline, missionModel);
+    }
+  }
+
+  /** Compute a set of results from the current state of simulation. */
+  // TODO: Move result extraction out of the SimulationEngine.
+  //   The Engine should only need to stream events of interest to a downstream consumer.
+  //   The Engine cannot be cognizant of all downstream needs.
+  // TODO: Whatever mechanism replaces `computeResults` also ought to replace `isTaskComplete`.
+  // TODO: Produce results for all tasks, not just those that have completed.
+  //   Planners need to be aware of failed or unfinished tasks.
+  public static SimulationResults computeResults(
+      final SimulationEngine engine,
+      final Instant startTime,
+      final Duration elapsedTime,
+      final Topic<ActivityInstanceId> activityTopic,
+      final TemporalEventSource timeline,
+      final MissionModel<?> missionModel
+  ) {
+
+    // Collect per-task information from the event graph.
+    final var taskInfo = new TaskInfo();
```

Comment:
Let's move `TaskInfo` out of the engine as well - I think it should live together with `computeResults`.

Reply:
Definitely agree, will work on that change!